src/is/L42/perftests/PerfCounters.java

```diff
@@ -0,0 +1,66 @@
+package is.L42.perftests;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class PerfCounters {
+	
+	private static Map<String, AtomicInteger> counter = new HashMap<>();
+	private static boolean enabled = false;
+	
+	public static boolean isEnabled() {
+		return enabled;
+		}
+	public static void setEnabled(boolean enabled) {
+		PerfCounters.enabled = enabled;
+		}
+	public static void inc(String key) {
+		ensureExists(key);
+		counter.get(key).incrementAndGet();
+		}
+	public static void add(String key, int amt) {
+		ensureExists(key);
+		counter.get(key).addAndGet(amt);
+		}
+	public static void reset(String key) {
+		ensureExists(key);
+		counter.get(key).set(0);
+		}
+	public static Optional<Integer> get(String key) {
+	  if(counter.containsKey(key)) {
+	    return Optional.empty();
+	    } else {
```

Comment:
There are better map method to do this with a single access (not just for the speed, it is irrelevant here, but to keep a consistent map-style